pycid/export/gambit.py

```diff
@@ -0,0 +1,223 @@
+                    cur_node.append_move(cur_infoset)
+                # else create a new infoset
+                else:
+                    cur_infoset = cur_node.append_move(player, len(actions))
+                    # label with the node for easy reference
+                    cur_infoset.label = node
+                    # add to infosets
+                    parents_to_infoset[parents_actions_tuple] = cur_infoset
+                # add state info
+                for action_idx, action in enumerate(actions):
+                    cur_infoset.actions[action_idx].label = str(action)
+                    state_info = node_idx_to_state[node_idx].copy()
+                    state_info.update({node: action})
+                    node_idx_to_state[node_idx + (action_idx,)] = state_info
+            else:
+                # otherwise is a chance node
+                factor = macid.query([node], context=parents_actions)
+                move = cur_node.append_move(game.players.chance, factor.cardinality)
+                # add state info
+                actions = macid.model.domain[node]
+                for action_idx, prob in enumerate(factor.values):
+                    move.actions[action_idx].label = str(actions[action_idx])
+                    move.actions[action_idx].prob = pygambit.Decimal(prob)
+                    state_info = node_idx_to_state[node_idx].copy()
+                    state_info.update({node: actions[action_idx]})
+                    node_idx_to_state[node_idx + (action_idx,)] = state_info
+
+    _add_payoffs(macid, game, range_num_children, node_idx_to_state, agents_in_sg)
+
+    return game, parents_to_infoset
+
+
+def macid_to_gambit_file(macid: MACIDBase, filename: str = "macid.efg") -> bool:
+    """Converts MACID to a ".efg" file for use with GAMBIT GUI.
+    Args:
+    - macid: The MACID object to convert to a pygambit EFG.
+    - filename: The filename to save the EFG to (default: macid.efg)."""
+    game, _ = macid_to_efg(macid)
+    with open(filename, "w") as f:
+        f.write(game.write())
+    print("\nGambit .efg file has been created from the macid")
+
+    return True
+
+
+def behavior_to_cpd(
+    macid: MACIDBase,
+    parents_to_infoset: Mapping[Tuple[Hashable, Tuple[Tuple[str, Any], ...]], pygambit.Infoset],
+    behavior: pygambit.lib.libgambit.MixedStrategyProfile,
+    decisions_in_sg: Union[KeysView[str], Set[str]] = None,
+) -> List[StochasticFunctionCPD]:
+    """Convert a pygambit behavior strategy to list of CPDs for each decision node.
+    Args:
+    - macid: The MACID object relating to the behavior strategy.
+    - parents_to_infoset: A mapping from (agent, (parent, instantiation), ...) to pygambit infoset.
+    - behavior: The pygambit behavior strategy.
+    - decisions_in_sg: The decisions in the subgame. If None, use all decisions in the MACID.
+    Returns:
+    - cpds: A list of CPDs for each decision node.
+    """
+
+    def _decimal_from_fraction(item: Any) -> float:
```

Comment:
kind of weird that you have to do this...

Reply:
my bad - actually we don't need to do this